oauthenticator/oauth2.py

```diff
@@ -644,6 +675,38 @@ async def token_to_user(self, token_info):
 
         return await self.fetch(req, "Fetching user info...")
 
+    def get_access_token_creation_date(self, token_info):
+        """
+        Returns the access token creation date, in seconds (Unix epoch time).
+
+        Example: 1679994631
+
+        Args:
+            token_info: the dictionary returned by the token request (exchanging the OAuth code for an Access Token)
+
+        Returns:
+            creation_date: a number representing the access token creation date, in seconds (Unix epoch time)
+
+        Called by the :meth:`oauthenticator.OAuthenticator.build_auth_state_dict`
+        """
+        return token_info.get("created_at", time.time())
+
+    def get_access_token_lifetime(self, token_info):
+        """
+        Returns the access token lifetime, in seconds.
+
+        Example: 7200
+
+        Args:
+            token_info: the dictionary returned by the token request (exchanging the OAuth code for an Access Token)
+
+        Returns:
+            lifetime: a number representing the access token lifetime, in seconds
+
+        Called by the :meth:`oauthenticator.OAuthenticator.build_auth_state_dict`
+        """
+        return token_info.get("expires_in", self.access_token_expiration)
```

Comment:
Need specific methods to retrieve refresh-related informations, as other providers may need custom logic.

But what if the required fields are provider via response headers ? May need provider implementation to see if that's relevant.